nodejs/week2/app.js

```diff
@@ -0,0 +1,75 @@
+const express = require("express");
+const app = express();
+const fs = require('fs')
+const bodyParser = require('body-parser');
+const documents = require('./documents.json');
+
+const port = process.env.PORT || 3000;
+
+// Support parsing JSON requests
+app.use(express.json());
+
+app.get("/", (req, res) => {
+    res.send("This is a search engine");
+});
+
+//GET /search
+app.get('/search', (req, res) => {
+    const q = req.query.q;
+    if (!q) {
+        return res.json(documents);
+    }
+
+    const searchSearch = documents.filter(doc => {
+        return Object.values(doc).some(value => {
+            if (typeof value === 'string') {
+                return value.toLocaleLowerCase().includes(q.toLocaleLowerCase());
+            }
+            return false;
+        });
+    });
+    return res.json(searchSearch);
+});
+
+//GET /documents/:id
+app.get("/documents/:id", (req, res) => {
+    documents = req.params.id;
+})
+
+//POST /search
+app.use(bodyParser.json());
```

Comment:
Would rather this was in the top of the file along with the other initialisation code